doc/source/conf.py

```diff
@@ -416,3 +418,55 @@ def linkcode_resolve(domain, info):
 SphinxDocString._str_examples = _rng_html_rewrite(
     SphinxDocString._str_examples
 )
+class LegacyDirective(Directive):
```

Comment:
Can be done in a follow up, but I would move this in its own file to not mix it with the config itself. Otherwise we are missing 2 spaces before it.
```suggestion


class LegacyDirective(Directive):
```

Reply:
Adding the spaces for now but happy to move it to another file in a follow up, together with the extra docs you mentioned below!